### ISSUE TITLE
Oncoprint takes long time to load, seems to be waiting for OncoKB

### ISSUE THREAD

**Issue by pieterlukasse** [cBioPortal/cbioportal#3046]
Many times oncoprint stalls and takes a long time to resume loading. It stops at "Downloading" and the call to http://www.cbioportal.org/genie/api-legacy/proxy/oncokb shows as "pending". 

I think it should continue loading. Perhaps it is waiting now because it needs OncoKB information to decide on driver/passenger coloring? I would like to suggest a change where the oncoprint code just continues and let the oncoKB response come in in parallel. The coloring could be done at the end just before final rendering.

**Comment by pieterlukasse** [comment]
@schultzn @jjgao this is the issue we briefly talked about friday.

**Comment by jjgao** [comment]
@pieterlukasse could you paste the query here?

We cache queries on the oncokb side, so it may be slow after we restart oncokb server, but usually the performance is ok once it's cached.

**Comment by pieterlukasse** [comment]
@jjgao here you go: http://www.cbioportal.org/genie/index.do?session_id=59a58b12498e5df2e2949722

**Comment by pieterlukasse** [comment]
@jjgao maybe good to look at it with tools like Fiddler. Here some images:

![image](https://user-images.githubusercontent.com/2900303/29830578-fb9f875c-8ce2-11e7-919d-c0d3a965789a.png)

a lot of time is spent in `fetch` operations (new API):

![image](https://user-images.githubusercontent.com/2900303/29830685-37c68e10-8ce3-11e7-91b5-1046db7fb02a.png)

**Comment by jjgao** [comment]
We will change the data fetching and API end point when refactoring -- hopefully we can address the performance issue.

The oncokb performance seems ok?

**Comment by pieterlukasse** [comment]
@jjgao the call to oncoKB in my figure above seems to not take too long, but my main issue with the OncoKB call is that all code just waits for it to return (see the last calls on the bottom right of the figure: they only start after oncokb is back, but could be probably in parallel).

**Comment by jjgao** [comment]
@pieterlukasse that's a good point. Once the mutation data is loaded, we should start to load oncokb data. @adamabeshouse would that be possible?

**Comment by adamabeshouse** [comment]
@jjgao I think we covered the issue in person, but to restate it here: if we render then automatically query for oncokb then rerender, that might be a jarring UX. Another solution would be to not automatically load oncokb, but I know thats probably not ideal. Is this still an issue after @zhx828 's optimizations?

**Comment by jjgao** [comment]
@adamabeshouse agreed about jarring ux.

I think there is some room we can improve oncokb. @zhx828 we probably need an API for pull oncogenicity data only (not levels). Would that potentially improve the performance?

**Comment by zhx828** [comment]
@jjgao absolutely. I'm actually working on that. Similar to genes API which should allow user specify which field to be returned.

**Comment by adamabeshouse** [comment]
is there still a performance problem or can we close this?

**Comment by zhx828** [comment]
@adamabeshouse I think after the last optimization, it should be good enough to close the ticket. I have the filter based on field specified in the request worked out in the upcoming release of OncoKB. I will open another ticket to use that feature.
